cyclient/cycode_token_based_client.py

```diff
@@ -29,23 +29,23 @@ def refresh_api_token_if_needed(self) -> None:
             self.refresh_api_token()
 
     def refresh_api_token(self) -> None:
-        auth_response = requests.post(f"{self.api_url}/api/v1/auth/api-token",
-                                      json={
-                                          'clientId': self.client_id,
-                                          'secret': self.client_secret
-                                      })
-        auth_response.raise_for_status()
-
+        auth_response = self.post(
+            url_path=f'api/v1/auth/api-token',
+            body={'clientId': self.client_id, 'secret': self.client_secret}
+        )
         auth_response_data = auth_response.json()
+
         self._api_token = auth_response_data['token']
-        self._expires_in = arrow.utcnow().shift(
-            seconds=auth_response_data['expires_in'] * 0.8)
+        self._expires_in = arrow.utcnow().shift(seconds=auth_response_data['expires_in'] * 0.8)
 
-    def get_request_headers(self, additional_headers: dict = None):
+    def get_request_headers(self, additional_headers: dict = None) -> dict:
         headers = super().get_request_headers(additional_headers=additional_headers)
-        headers = self._add_auth_header(headers)
+
+        if not self.lock.locked():
```

Comment:
what was the race condition that it comes to solve?

Reply:
this solves the problem when we are attempting to fetch the token for the first time. when we request it we don't have an access token yet. so we need to perform the request without Auth header. without the lock here is the deadlock